irohad/multi_sig_transactions/transport/impl/mst_transport_grpc.cpp

```diff
@@ -150,26 +155,33 @@ void MstTransportGrpc::subscribe(
 void MstTransportGrpc::sendState(const shared_model::interface::Peer &to,
                                  ConstRefState providing_state) {
   log_->info("Propagate MstState to peer {}", to.address());
-  sendStateAsyncImpl(to, providing_state, my_key_, *async_call_);
+  sendStateAsyncImpl(
+      to, providing_state, my_key_, *async_call_, client_creator_);
 }
 
 void iroha::network::sendStateAsync(
     const shared_model::interface::Peer &to,
     ConstRefState state,
     const shared_model::crypto::PublicKey &sender_key,
     AsyncGrpcClient<google::protobuf::Empty> &async_call) {
-  sendStateAsyncImpl(
-      to, state, shared_model::crypto::toBinaryString(sender_key), async_call);
+  sendStateAsyncImpl(to,
+                     state,
+                     shared_model::crypto::toBinaryString(sender_key),
+                     async_call,
+                     nullptr);
 }
 
-void sendStateAsyncImpl(const shared_model::interface::Peer &to,
-                        ConstRefState state,
-                        const std::string &sender_key,
-                        AsyncGrpcClient<google::protobuf::Empty> &async_call) {
-  std::unique_ptr<transport::MstTransportGrpc::StubInterface> client =
-      transport::MstTransportGrpc::NewStub(grpc::CreateChannel(
-          to.address(), grpc::InsecureChannelCredentials()));
-
+void sendStateAsyncImpl(
+    const shared_model::interface::Peer &to,
+    ConstRefState state,
+    const std::string &sender_key,
+    AsyncGrpcClient<google::protobuf::Empty> &async_call,
+    std::function<std::unique_ptr<transport::MstTransportGrpc::StubInterface>(
+        const shared_model::interface::Peer &)> client_creator) {
+  auto client = (client_creator == nullptr)
```

Comment:
As I understand, `nullptr` is used to reduce the changes in the pull request. In general this should not be neither boost::none, nor nullptr, but it would require to refactor the test. I suggest to add a TODO to fix the tests instead.

Reply:
I have changed this to boost::optional so that there is no ternary operator in client declaration.